account_statement_completion_compassion/model/statement.py

```diff
@@ -259,6 +268,26 @@ def _generate_invoice(self, cr, uid, st_line, partner, context=None):
                     wf_service.trg_validate(
                         uid, 'account.invoice', invoice_id, 'invoice_open',
                         cr)
+                elif product.name == GIFT_TYPES[0]:
+                    # Set date of invoice two months before child's birthdate
+                    child_birthdate = res.get('child_birthdate')
+                    if child_birthdate:
+                        inv_date = datetime.strptime(st_line['date'], DF)
+                        birthdate = datetime.strptime(child_birthdate, DF)
+                        new_date = inv_date
```

Comment:
In fact it is not as easy as the comment suggest it... if the payment is made less than three months after the birthday, the invoice_date will be at the same date of the payment.
In other situation, it is set two months before the birthday.

Moreover, child_birthdate has a very old year, we should keep the current year for the invoice, or next year if birthday is in future.